src/write.c

```diff
@@ -1818,7 +1818,11 @@ static avifResult avifEncoderWriteMediaDataBox(avifEncoder * encoder,
             // Deduplication - See if an identical chunk to this has already been written
             if (item->encodeOutput->samples.count > 0) {
                 avifEncodeSample * sample = &item->encodeOutput->samples.sample[0];
-                chunkOffset = avifEncoderFindExistingChunk(s, mdatStartOffset, sample->data.data, sample->data.size);
+                size_t total_size = sample->data.size;
+                for (uint32_t sampleIndex = 1; sampleIndex < item->encodeOutput->samples.count; ++sampleIndex) {
+                    total_size += item->encodeOutput->samples.sample[sampleIndex].data.size;
```

Comment:
Is the data contiguous?

Suggestion:

```c
AVIF_CHECKERR(
  item->encodeOutput->samples.sample[sampleIndex].data.data ==
    item->encodeOutput->samples.sample[0].data.data + total_size,
  AVIF_RESULT_UNKNOWN_ERROR);
```

(not just an assertion, for extra security)

Reply:
Well, the data is written contiguously a few lines below but I'll add a check.
The problem happened starting with https://github.com/AOMediaCodec/libavif/pull/463

Reply:
Actually, we do not care right? The stream is already formed so as long as we have a match of size total_size, we are good and the deduplication works.

Reply:
I missed it's on the input, that makes sense. Done